### PR TITLE
feat(action logs): download archived log for finished job runs

### PR DESCRIPTION
**Pull request by caoer**
## What

Makes `cocli action logs` work for **finished** job runs, not just running ones — transparently, with no new flag.

- **Running job:** stream live pod logs (unchanged).
- **Finished job (pod/workflow GC'd):** the server (apiary) returns a presigned URL for the archived log in `LogJobRunResponse.log_download_uri`; cocli downloads it and prints the contents.
- **Not-yet-started job (queued/scheduling):** no pod and no archive yet — with `-f`, poll and wait for it to start; without `-f`, report the state and exit cleanly (instead of appearing to "finish" with no output). _(addresses review feedback)_
- **Finished job with no archived log** (no output, or archival off when it ran): report `No logs available for this job run.` rather than a raw 404. _(found via e2e monkey testing)_

## Docs
Added a `Long` description + `Examples` to `cocli action logs -h` covering the live-vs-archived behavior and the `-j` / `--node` / `-f` flags.

## Tests
Extracted per-message handling into `handleLogMessage` and added unit tests: `handleLogMessage` (live line / archived download), `printArchivedLog` (success / 404 / 500), `awaitJobRunStart` (already-started / no-follow / poll-until-running / GetJobRun error / canceled), `jobRunNotStarted`. CI: build, codecov/project, ubuntu+macos tests all green.

## Dependencies
- coscene-openapi #69 (`log_download_uri`) — merged; go.mod bumped to that SDK snapshot.
- apiary #81 — emits the presigned URL (paired).
- integration-tests #6 — self-reproducing Argo log archival for the e2e env.

## Validated e2e
On the remote-dev-container e2e env: `cocli action logs <finished-run> -j 0` → apiary emitted a **publicly-accessible** presigned URL → cocli downloaded + printed the archived encoder log (exit 0). Edge cases + 5× concurrent exercised via a monkey test.

### CODE REVIEW

File: pkg/cmd/action/logs.go [hunk 1]
```diff
@@ -197,11 +203,48 @@ func streamOnce(ctx context.Context, cli api.JobRunInterface, jobRunName, node s
 	defer func() { _ = stream.Close() }()
 
 	for stream.Receive() {
```
Comment (juchaosong) [inline pkg/cmd/action/logs.go:305]: 如果cocli logs的时候，这个job还没有开始运行，现在的逻辑好像就直接退出了

Comment (caoer) [reply pkg/cmd/action/logs.go:305]: 好建议，已修复 (d16a6dc)。

之前 queued/scheduling 状态的 job run 既没有 pod 可以 stream，也没有归档日志可下载，所以 stream 直接空结束、看起来像"已完成无输出"。

现在的处理：
- 带 -f：轮询 job run 状态，等待它进入 running/终态后再开始 stream/下载；
- 不带 -f：打印当前状态提示（如 "Job run has not started yet (state: QUEUED). Pass -f to wait for logs."）并干净退出，而不是报错或假装完成。

running / 终态（succeeded/failed/aborted）的行为不变（实时 stream / 归档日志下载）。已加单测覆盖。

**Review by juchaosong — APPROVED** [review]
